### PR TITLE
Create Classes and autoload

### PR DESCRIPTION
**Pull request by sergynyo**
https://gmslabs.atlassian.net/browse/HP-83

### CODE REVIEW

File: Hyber/Autoload.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Hyber;
+
+class Autoload
```
Comment (arodiss) [inline Hyber/Autoload.php]: @sergynyo 
no need, this is managed by Composer - you only have to specify namespace root.
Check any existing library on GitHub

File: Hyber/Sender/Message.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function __construct($phoneNumber)
+    {
+        $this->phoneNumber = $phoneNumber;
+    }
+
+    /**
+     * @param Push $push
+     */
+    public function setPush(Push $push)
```
Comment (arodiss) [inline Hyber/Sender/Message.php]: @sergynyo 
please match the API

File: Hyber/Sender/Message.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    
+    /** @var Sms */
+    private $sms;
+
+    /**
+     * @param string $phoneNumber
+     */
+    public function __construct($phoneNumber)
+    {
+        $this->phoneNumber = $phoneNumber;
```
Comment (arodiss) [inline Hyber/Sender/Message.php]: @sergynyo 
please match the API

File: Hyber/Autoload.php (hunk 1 above)
Comment (sergynyo) [inline Hyber/Autoload.php]: @arodiss 
I took as a basis the
https://github.com/PHPOffice/PHPExcel
https://github.com/serj1chen/nova-poshta-sdk-php

If user dont use composer, he can take autoload

Comment (arodiss) [inline Hyber/Autoload.php]: @sergynyo 
You may leave custom autoloader, but you still have to implement composer-compatible autoload too

File: Hyber/Sender/MessageSender.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * @param Message   $message
+     * @param \DateTime $startTime
+     * @return bool
+     */
+    public function send(Message $message, $startTime = null)
+    {
+        // TODO: implement convert object to array and send to Java
+        return true;
```
Comment (arodiss) [inline Hyber/Sender/MessageSender.php]: @sergynyo 
Please implement 
Please add required dependency to `composer.json`

File: Hyber/Autoload.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+            spl_autoload_register('__autoload');
+        }
+
+        return spl_autoload_register(array('\Hyber\Autoload', 'load'));
+    }
+
+    public static function load($className)
+    {
+        $className = str_replace('Hyber\\', '', $className);
+        $className = HYBER_PATH_SDK . $className . '.php';
```
Comment (arodiss) [inline Hyber/Autoload.php]: @sergynyo 
`HYBER_PATH_SDK` is not defined

File: Hyber/MessageSender.php [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    private $password;
+
+    /** @var integer */
+    private $identifier;
+
+    /** @var string */
+    private $alphaName;
+
+    /** @var string */
+    private $apiHost;
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please replace this with constant with value `https://api.hyber.im`

File: Hyber/MessageSender.php [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+
+    /** @var string */
+    private $callbackUrl;
+
+    /**
+     * @param string  $login
+     * @param string  $password
+     * @param integer $identifier
+     * @param string  $alphaName
+     * @param string  $apiHost
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please remove this param

File: README.md [hunk 8]
```diff
@@ -7,7 +7,7 @@ Easy way to integrate PHP-powered system with Hyber platform
 ```PHP
 // First, you need to create service that will send your messages.
 // All this parameters are mandatory. They are provided for each Hyber customer and rarely change  
-$sender = new Hyber\MessageSender($login, $password, $identifier, $alphaName);
+$sender = new Hyber\MessageSender($login, $password, $identifier, $alphaName, $apiHost);
```
Comment (arodiss) [inline README.md]: @sergynyo 
Please revert this change

File: composer.json [hunk 9]
```diff
@@ -6,7 +6,8 @@
   "homepage": "https://hyber.ch",
   "license": "MIT",
   "require": {
-    "php": ">=5.3.9"
+    "php": ">=5.3.9",
+    "guzzlehttp/guzzle": "6.*"
```
Comment (arodiss) [inline composer.json]: @sergynyo 
Please use HTTP Plug: https://github.com/php-http/httplug
Please don't forget to update type hints

File: README.md [hunk 10]
```diff
@@ -3,22 +3,64 @@ Easy way to integrate PHP-powered system with Hyber platform
 
 [![Build Status](https://travis-ci.org/Incuube/Hyber-SDK-PHP.svg)](https://travis-ci.org/Incuube/Hyber-SDK-PHP)
 
+## Install
+
+#### Via Composer
+
+Create a file `composer.json` in your project root:
```
Comment (arodiss) [inline README.md]: @sergynyo 
This is done via CLI:
"composer require incuube/sdk-hyber"

File: README.md [hunk 11]
```diff
@@ -3,22 +3,64 @@ Easy way to integrate PHP-powered system with Hyber platform
...
+          "require": {
+              "Incuube/sdk-hyber": "*"
+          }
+      }
+
+#### Download
+
+Clone a repository or download
+
+      git clone git://github.com/Incuube/Hyber-SDK-PHP
```
Comment (arodiss) [inline README.md]: @sergynyo 
C'mon, one never clones vendors from git, they only need source code
```
wget https://github.com/Incuube/Hyber-SDK-PHP/archive/master.zip
```

File: README.md [hunk 12]
```diff
@@ -3,22 +3,64 @@ Easy way to integrate PHP-powered system with Hyber platform
...
+        'Accept' => 'application/json',
+    ],
+    'auth' => [
+        $login,
+        $password,
+    ],
+];
+
+$guzzle = new GuzzleClient($config);
+$adapter = new GuzzleAdapter($guzzle);
```
Comment (arodiss) [inline README.md]: @sergynyo 
Please add namespace for this two classes

File: README.md [hunk 13]
```diff
@@ -3,22 +3,64 @@ Easy way to integrate PHP-powered system with Hyber platform
...
-// Second, you need to create and configure message instance
+// Third, you need to create and configure message instance
 // Parameters in constructor are mandatory, parameters in setters are not
 $message = new Hyber\Message($phoneNumber);
 $message->setExtraId($mySystem->getMessageId()); //some identifier from external system
 $message->setTag('campaign'); //on Hyber portal you can filter statistics by tag
 $message->setIsPromotional(true); //whether or not your message is promotopnal (advertising)
 
-// Third, you need to configure channels with which your message will be sent
+// Fourthly, you need to configure channels with which your message will be sent
```
Comment (arodiss) [inline README.md]: @sergynyo 
"Fourth"

File: composer.json [hunk 14]
```diff
@@ -6,7 +6,11 @@
   "homepage": "https://hyber.ch",
   "license": "MIT",
   "require": {
-    "php": ">=5.3.9"
+    "php": ">=5.3.9",
+    "guzzlehttp/guzzle": "6.*",
```
Comment (arodiss) [inline composer.json]: @sergynyo 
Please remove this requirement - if we force specifi version of Guzzle, we don't need HttpPlug

File: Hyber/ApiClient.php [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory
+     */
+    private $requestFactory;
+
+    public function __construct(HttpClient $httpClient, RequestFactory $requestFactory)
```
Comment (arodiss) [inline Hyber/ApiClient.php]: @sergynyo 
docblock

File: Hyber/ApiClient.php [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * @param $uri
+     * @param $payload
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public function apiCall($uri, $payload)
+    {
+        $request = $this->requestFactory->createRequest('POST', $uri, ['content-type' => 'application/json'], $payload);
+        return $this->httpClient->sendRequest($request);
```
Comment (arodiss) [inline Hyber/ApiClient.php]: @sergynyo 
NL before `return`

File: Hyber/Message.php [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function addPush(Push $push)
+    {
+        $this->push = $push;
+    }
+    
+    /**
+     * @return Viber
+     */
+    public function getViber()
```
Comment (arodiss) [inline Hyber/Message.php]: @sergynyo 
Please remove this method

File: Hyber/Message.php [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function __construct($phoneNumber)
+    {
+        $this->phoneNumber = $phoneNumber;
+    }
+
+    /**
+     * @return Push
+     */
+    public function getPush()
```
Comment (arodiss) [inline Hyber/Message.php]: @sergynyo 
Please remove this method

File: Hyber/Message.php [hunk 19]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function addViber(Viber $viber)
+    {
+        $this->viber = $viber;
+    }
+
+    /**
+     * @return Sms
+     */
+    public function getSms()
```
Comment (arodiss) [inline Hyber/Message.php]: @sergynyo 
Please remove this method

File: Hyber/Message.php [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function setPhoneNumber($phoneNumber)
+    {
+        $this->phoneNumber = $phoneNumber;
+    }
+
+    /**
+     * @return int
+     */
+    public function getExtraId()
```
Comment (arodiss) [inline Hyber/Message.php]: @sergynyo 
Please remove this method

File: Hyber/Message.php [hunk 21]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function setExtraId($extraId)
+    {
+        $this->extraId = $extraId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTag()
```
Comment (arodiss) [inline Hyber/Message.php]: @sergynyo 
Please remove this method

File: Hyber/Message.php [hunk 22]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    public function setTag($tag)
+    {
+        $this->tag = $tag;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getIsPromotional()
```
Comment (arodiss) [inline Hyber/Message.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Push.php [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    public function setText($text)
+    {
+        $this->text = $text;
+    }
+
+    /**
+     * @return int
+     */
+    public function getTtl()
```
Comment (arodiss) [inline Hyber/Message/Push.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Push.php [hunk 24]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    public function getText()
+    {
+        return $this->text;
+    }
+
+    /**
+     * @param string $text
+     */
+    public function setText($text)
```
Comment (arodiss) [inline Hyber/Message/Push.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Push.php [hunk 25]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    public function getTtl()
+    {
+        return $this->ttl;
+    }
+
+    /**
+     * @param int $ttl
+     */
+    public function setTtl($ttl)
```
Comment (arodiss) [inline Hyber/Message/Push.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Push.php [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    public function addImage($image)
+    {
+        $this->image = $image;
+    }
+
+    /**
+     * @return string
+     */
+    public function getImage()
```
Comment (arodiss) [inline Hyber/Message/Push.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Push.php [hunk 27]
```diff
@@ -0,0 +1,96 @@
...
+        $this->button = [
+            'caption' => $caption,
+            'link' => $link
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function getButton()
```
Comment (arodiss) [inline Hyber/Message/Push.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Sms.php [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+    public function __construct($text, $ttl)
+    {
+        $this->text = $text;
+        $this->ttl = $ttl;
+    }
+
+    /**
+     * @return string
+     */
+    public function getText()
```
Comment (arodiss) [inline Hyber/Message/Sms.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Sms.php [hunk 29]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        return $this->text;
+    }
+
+    /**
+     * @param string $text
+     */
+    public function setText($text)
```
Comment (arodiss) [inline Hyber/Message/Sms.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Sms.php [hunk 30]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        $this->text = $text;
+    }
+
+    /**
+     * @return int
+     */
+    public function getTtl()
+    {
+        return $this->ttl;
```
Comment (arodiss) [inline Hyber/Message/Sms.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Viber.php [hunk 31]
```diff
@@ -0,0 +1,115 @@
...
+    public function getText()
+    {
+        return $this->text;
+    }
+
+    /**
+     * @param string $text
+     */
+    public function setText($text)
+    {
```
Comment (arodiss) [inline Hyber/Message/Viber.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Viber.php [hunk 32]
```diff
@@ -0,0 +1,115 @@
...
+        $this->text = $text;
+    }
+
+    /**
+     * @return int
+     */
+    public function getTtl()
+    {
+        return $this->ttl;
```
Comment (arodiss) [inline Hyber/Message/Viber.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Viber.php [hunk 33]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function addImage($image)
+    {
+        $this->image = $image;
+    }
+
+    /**
+     * @return string
+     */
+    public function getImage()
```
Comment (arodiss) [inline Hyber/Message/Viber.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Viber.php [hunk 34]
```diff
@@ -0,0 +1,115 @@
...
+            'caption' => $caption,
+            'link' => $link
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function getButton()
+    {
```
Comment (arodiss) [inline Hyber/Message/Viber.php]: @sergynyo 
Please remove this method

File: Hyber/Message/Viber.php [hunk 35]
```diff
@@ -0,0 +1,115 @@
...
+    public function addIosExpirityText($iosExpirityText)
+    {
+        $this->iosExpirityText = $iosExpirityText;
+    }
+
+    /**
+     * @return string
+     */
+    public function getIosExpirityText()
+    {
```
Comment (arodiss) [inline Hyber/Message/Viber.php]: @sergynyo 
Please remove this method

File: Hyber/MessageSender.php [hunk 36]
```diff
@@ -0,0 +1,275 @@
...
+use GuzzleHttp\Client;
+use Hyber\Message\Push;
+use Hyber\Message\Sms;
+use Hyber\Message\Viber;
+use Hyber\Response\ErrorResponse;
+use Hyber\Response\SuccessResponse;
+
+class MessageSender
+{
+    const MESSAGE_CREATE_PATH = "/api/%s/messages/create";
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
`%s/messages/create`

File: Hyber/MessageSender.php [hunk 37]
```diff
@@ -0,0 +1,275 @@
...
+    const CODE_PHONE_NUMBER_INCORRECT = 1154;
+    const API_HOST = "https://api.hyber.im/http-receiver";
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
`https://api.hyber.im`

File: Hyber/MessageSender.php [hunk 38]
```diff
@@ -0,0 +1,275 @@
...
+        }
+
+        $data = $this->convertMessageToArray($message, $startTime);
+        if ($data instanceof ErrorResponse) {
+            return $data;
+        }
+
+        try {
+            return $this->doSendMessage($data);
+        } catch (\Exception $e) {
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
This try-catch does nothing, remove it

File: Hyber/MessageSender.php [hunk 39]
```diff
@@ -0,0 +1,275 @@
...
+                $error = new ErrorResponse();
+                $error->setErrorText("Invalid response detected");
+                return $error;
+            }
+
+            if (isset($response['error_code']) && isset($response['error_text'])) {
+                $error = new ErrorResponse();
+                $error->setErrorCode($response['error_code']);
+                $error->setErrorText($response['error_text']);
+                return $error;
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
NL before return

File: Hyber/MessageSender.php [hunk 40]
```diff
@@ -0,0 +1,275 @@
...
+            }
+
+            if (isset($response['message_id'])) {
+                return new SuccessResponse($response['message_id']);
+            }
+        } catch (\Exception $e) {
+            throw $e;
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
this try-catch does nothing, remove it

File: Hyber/MessageSender.php [hunk 41]
```diff
@@ -0,0 +1,275 @@
...
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param $phoneNumber
+     * @return string|null
+     */
+    public function validatePhoneNumber($phoneNumber)
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
protected

**Comment by sergynyo** [comment]
@arodiss 
done

File: Hyber/Message/Push.php [hunk 42]
```diff
@@ -0,0 +1,48 @@
...
+    public $text;
+    
+    /** @var integer */
+    public $ttl;
+
+    /** @var string */
+    public $image;
+
+    /** @var array */
+    public $button = [];
```
Comment (arodiss) [inline Hyber/Message/Push.php]: @sergynyo 
Should not be public (same for 1000 other places)

**Comment by sergynyo** [comment]
@arodiss 
done

File: Hyber/MessageSender.php [hunk 43]
```diff
@@ -0,0 +1,267 @@
...
+            $error = new ErrorResponse();
+            $error->setErrorCode($response['error_code']);
+            $error->setErrorText($response['error_text']);
+
+            return $error;
+        }
+
+        if (isset($response['message_id'])) {
+            return new SuccessResponse($response['message_id']);
+        }
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please add final item (considered as error) if none of checks fired. For example, if we got response `{"foo": "bar"}`

File: Hyber/MessageSender.php [hunk 44]
```diff
@@ -0,0 +1,267 @@
...
+            'texts' => $channels['texts'],
+            'channels' => $channels['channels'],
+            'phone_number' => $phone,
+            'extra_id' => $message->getExtraId(),
+            'callback_url' => $this->callbackUrl,
+            'alpha_name' => $this->alphaName
+        ];
+
+        $tag = $message->getTag();
+        if ($tag) {
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
```PHP
if ($message->getTag()) {
    $data['tag'] =  $message->getTag();
}
```

File: Hyber/MessageSender.php [hunk 45]
```diff
@@ -0,0 +1,267 @@
...
+            $data['tag'] = $tag;
+        }
+
+        $startTime = $this->convertStartTime($startTime);
+        if ($startTime) {
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
```PHP
if ($startTime = $this->convertStartTime($startTime);) {
    $data['start_time'] = $startTime;
}
```

File: Hyber/MessageSender.php [hunk 46]
```diff
@@ -0,0 +1,267 @@
...
+        }
+
+        return $phone;
+    }
+
+    /**
+     * @param Message $message
+     * @return array
+     */
+    private function convertChannelsToArray(Message $message)
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please put this functionality to `Message` class

File: Hyber/MessageSender.php [hunk 47]
```diff
@@ -0,0 +1,267 @@
...
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param $phoneNumber
+     * @return string|null
+     */
+    private function validatePhoneNumber($phoneNumber)
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please put this functionality to `Message` class

File: Hyber/MessageSender.php [hunk 48]
```diff
@@ -0,0 +1,267 @@
...
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param $dateTime
+     * @return null|string
+     */
+    private function convertStartTime($dateTime)
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please put this functionality to `Message` class

File: Hyber/MessageSender.php [hunk 49]
```diff
@@ -0,0 +1,267 @@
...
+        } else {
+            return $dateTime;
+        }
+    }
+
+    /**
+     * @param Message $message
+     * @return array
+     */
+    private function buildChannelOptions(Message $message)
```
Comment (arodiss) [inline Hyber/MessageSender.php]: @sergynyo 
Please put this functionality to `Message` class

File: Hyber/Response/ErrorResponse.php [hunk 50]
```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Hyber\Response;
+
+class ErrorResponse
+{
+    /** @var integer */
+    private $httpCode;
```
Comment (arodiss) [inline Hyber/Response/ErrorResponse.php]: @sergynyo 
Please add default value 0

File: README.md [hunk 51]
```diff
@@ -3,22 +3,57 @@ Easy way to integrate PHP-powered system with Hyber platform
 
 [![Build Status](https://travis-ci.org/Incuube/Hyber-SDK-PHP.svg)](https://travis-ci.org/Incuube/Hyber-SDK-PHP)
 
+## Installaton
+
+#### Method 1 (composer)
```
Comment (arodiss) [inline README.md]: @sergynyo 
"Recommended: composer"

File: README.md [hunk 52]
```diff
@@ -3,22 +3,57 @@ Easy way to integrate PHP-powered system with Hyber platform
...
+ 
+      composer require incuube/sdk-hyber
+
+#### Method 2 (download)
```
Comment (arodiss) [inline README.md]: @sergynyo 
"Not recommended: download sources"
